code/datums/status_effects/xeno_buffs.dm

```diff
@@ -24,6 +27,335 @@
+
+/datum/status_effect/stacking/essence_link/tick()
+	var/remaining_health = link_target.maxHealth - (link_target.getBruteLoss() + link_target.getFireLoss())
+	var/heal_amount = round(link_target.maxHealth * (stacks * DRONE_ESSENCE_LINK_REGEN))
+	var/heal_remainder = round(max(0, heal_amount - link_target.getBruteLoss()))
+	var/plasma_cost = round(heal_amount * 2.5)
+	if(stacks < max_stacks && COOLDOWN_CHECK(src, attunement_increase))
+		add_stacks(1)
+	if(!was_within_range || remaining_health >= link_target.maxHealth)
+		return
+	if(link_owner.plasma_stored < plasma_cost)
+		if(!COOLDOWN_CHECK(src, plasma_warning))
+			return
+		link_owner.balloon_alert(link_owner, "No plasma for link")
+		link_target.balloon_alert(link_target, "No plasma for link")
+		COOLDOWN_START(src, plasma_warning, plasma_warning_cooldown)
+		return
+	link_target.adjustBruteLoss(-heal_amount)
+	link_target.adjustFireLoss(-heal_remainder)
+	link_owner.use_plasma(plasma_cost)
+
+/// Removes the status effect on death.
+/datum/status_effect/stacking/essence_link/proc/handle_death(datum/source)
+	SIGNAL_HANDLER
+	essence_link_action.end_ability()
+
+/// Toggles the link depending on whether the linked xenos are still in range or not.
+/datum/status_effect/stacking/essence_link/proc/handle_dist(datum/source)
+	SIGNAL_HANDLER
+	var/within_range = get_dist(link_owner, link_target) <= DRONE_ESSENCE_LINK_RANGE
+	if(within_range == was_within_range)
+		return
+	was_within_range = within_range
+	toggle_link(was_within_range)
+	link_owner.balloon_alert(link_owner, was_within_range ? ("Link reestablished") : ("Link faltering"))
+	link_target.balloon_alert(link_target, was_within_range ? ("Link reestablished") : ("Link faltering"))
+
+/// Shares the Resin Jelly buff with the linked xeno.
+/datum/status_effect/stacking/essence_link/proc/share_jelly(datum/source)
+	SIGNAL_HANDLER
+	var/mob/living/carbon/xenomorph/buff_owner
+	var/mob/living/carbon/xenomorph/buff_target
+
+	if(source == link_target)
+		buff_owner = link_target
+		buff_target = link_owner
+	else
+		buff_owner = link_owner
+		buff_target = link_target
+
+	buff_owner.balloon_alert(buff_owner, "Buff shared")
+	buff_target.balloon_alert(buff_target, "Buff shared")
+	buff_target.visible_message(span_notice("[buff_target]'s chitin begins to gleam with an unseemly glow..."), \
+		span_xenonotice("Through the essence link, [buff_owner] has shared their resin jelly with you."))
+	INVOKE_ASYNC(buff_target, /mob/living/carbon/xenomorph.proc/emote, "roar")
+	buff_target.apply_status_effect(STATUS_EFFECT_RESIN_JELLY_COATING)
+
+/// Shares heals with the linked xeno.
+/datum/status_effect/stacking/essence_link/proc/share_heal(datum/source, amount)
+	SIGNAL_HANDLER
+	var/mob/living/carbon/xenomorph/heal_owner
+	var/mob/living/carbon/xenomorph/heal_target
+
+	// Prevents passive regeneration from triggering this. Most values do not go past 60.
+	if(amount > -60)
```

Comment:
That is awfull

Reply:
Duplicate comment, the one under this one seems to be referencing the issue better.